WordPress/Classes/Services/AccountSettingsService.swift

```diff
@@ -37,91 +50,51 @@ class AccountSettingsService {
-            .just(.Stalled)
-            .delaySubscription(Defaults.stallTimeout, scheduler: self.scheduler)
-    }()
-
-    /// Performs a network refresh of settings and emits values with the refresh status.
-    ///
-    /// - When it's subscribed, it requests a refresh from the server
-    /// - If it takes more than `stallTimeout` to complete, it will emit a `.Stalled` value and continue waiting for the request to finish.
-    /// - If a networking error happens it doesn't emit a new value and will retry the request.
-    /// - If it reaches the maximum permitted number of retries it will emit an Error.
-    /// - If an error not related to networking happens, it will emit an Error.
-    /// - When the data is refreshed, it will emit an `.Idle` value and complete.
-    lazy private(set) var request: Observable<RefreshStatus> = {
-        let remoteSettings = self.remoteSettings.shareReplayLatestWhileConnected()
-        let stalledSettings = Observable.of(self.stalled, remoteSettings)
-            .merge()
-
-        return remoteSettings
-            .amb(stalledSettings)
-            .startWith(.Refreshing)
-    }()
-
-    /// Emits values when the refresh status changes.
-    ///
-    /// On subscription, this will start refreshing settings, polling each minute, while there's an internet connection.
-    /// Possible values:
-    /// - `.Refreshing` when it starts getting remote data.
-    /// - `.Stalled` when it's getting remote data and hasn't succeeded before `stallTimeout`.
-    /// - `.Offline` when there is no internet connection.
-    /// - `.Idle` when the request was successful and it's waiting for the polling interval.
-    /// - An error when the request couldn't complete. It will stop retrying.
-    lazy var refresh: Observable<RefreshStatus> = {
-        // Copy request to avoid capture of self in closure
-        let request = self.request
-
-        // Convert to a polling request
-        let polling = Observable<Int>
-            .interval(Defaults.pollingInterval, scheduler: self.scheduler)
-            .startWith(0)
-            .flatMapLatest({ _ in request })
-
-        // Enable only when reachable, otherwise emit .Offline
-        return self.reachable.flatMapLatest({ reachable -> Observable<RefreshStatus> in
-            if reachable {
-                return polling
-            } else {
-                return Observable.just(.Offline)
+                if error.domain == NSURLErrorDomain && count < Defaults.maxRetries {
+                    self.getSettingsAttempt(count: count + 1)
+                } else {
+                    self.status = .Failed
+                }
             }
-        })
-    }()
+        )
+    }
+
+    func refreshSettings() {
+        guard status == .Idle || status == .Failed else {
+            return
+        }
+        status = .Refreshing
+        getSettingsAttempt()
+        stallTimer = NSTimer.scheduledTimerWithTimeInterval(Defaults.stallTimeout,
```

Comment:
We had that with RxSwift and I removed it because while I think it's an interesting thing to add in general, I don't think it adds much value for profile/settings, and it adds a lot more complexity.